deploy/role.yaml

```diff
@@ -120,3 +121,12 @@ rules:
   - create
   - delete
   - update
+  - list
+  - watch
```

Comment:
This seems like the wrong thing to do. Why do we need to list and watch secrets for this? Shouldn't we just add another rbac grouping for pods if that is what's needed?

Reply:
`watch` is not needed. I just reconfirmed with the SDK team too. `list` would be required and is sufficient!